sunkit_image/asda.py

```diff
@@ -11,410 +11,379 @@
     """
-    Version 2.0 of Automated Swirl Detection Algorithm (ASDA).
+    Calculate ``gamma1`` and ``gamma2`` values of velocity field vx and vy.
 
-    References
+    Parameters
     ----------
-    * `ASDA Source Code <https://github.com/PyDL/ASDA>`__.
+    vx : `numpy.ndarray`
+        Velocity field in the x direction.
+    vy : `numpy.ndarray`
+        Velocity field in the y direction.
+    factor : `int`, optional
+        Magnify the original data to find sub-grid vortex center and boundary.
+        Default value is 1.
+    r : `int`, optional
+        Maximum distance of neighbor points from target point.
+        Default value is 3.
+
+    Returns
+    -------
+    `tuple`
+        A tuple in form of ``(gamma1, gamma2)``, where ``gamma1`` is useful in
+        finding vortex centers and ``gamma2`` is useful in finding vortex
+        edges.
     """
 
-    def __init__(self, vx, vy, r=3, factor=1):
-        """
-        Parameters
-        ----------
-        vx : `numpy.ndarray`
-            Velocity field in the x direction.
-        vy : `numpy.ndarray`
-            Velocity field in the y direction.
-        r : `int`, optional
-            Maximum distance of neighbor points from target point.
-            Default value is 3.
-        factor : `int`, optional
-            Magnify the original data to find sub-grid vortex center and boundary.
-            Default value is 1.
-
-        References
-        ----------
-        * Laurent Graftieaux, Marc Michard and Nathalie Grosjean.
-          Combining PIV, POD and vortex identification algorithms for the
-          study of unsteady turbulent swirling flows.
-          Meas. Sci. Technol. 12, 1422, 2001.
-          (https://doi.org/10.1088/0957-0233/12/9/307)
-        * Jiajia Liu, Chris Nelson, Robert Erdelyi.
-          Automated Swirl Detection Algorithm (ASDA) and Its Application to
-          Simulation and Observational Data.
-          Astrophys. J., 872, 22, 2019.
-          (https://doi.org/10.3847/1538-4357/aabd34)
-
-        """
-        if vx.shape != vy.shape:
-            raise ValueError("Shape of velocity field's vx and vy do not match")
-        if not isinstance(r, int):
-            raise ValueError("Keyword 'r' must be an integer")
-        if not isinstance(factor, int):
-            raise ValueError("Keyword 'factor' must be an integer")
-        self.vx = vx
-        self.vy = vy
-        self.dshape = np.shape(vx)
-        self.r = r
-        self.factor = factor
-
-    def gen_vel(self, i, j):
-        """
-        Given a point ``[i, j]``, generate a velocity field which contains a
-        region with a size of ``(2r+1) x (2r+1)`` centered at ``[i, j]`` from
-        the original velocity field ``self.vx`` and ``self.vy``.
+    if vx.shape != vy.shape:
+        raise ValueError("Shape of velocity field's vx and vy do not match")
+    if not isinstance(r, int):
+        raise ValueError("Keyword 'r' must be an integer")

```

Comment:
We want to make sure these are covered by tests.

Reply:
sure, I will do it.